### PR TITLE
Merge type stubs torch nn parallel

### PR DESCRIPTION
**Pull request by ringohoffman**
Fixes merge issue for #101528

In the above PR, `torch.nn.parallel.parallel_apply.get_a_var` was marked private to appease the [public interface linter](https://github.com/pytorch/pytorch/actions/runs/4999216467/jobs/8955582204#step:14:21666): https://github.com/pytorch/pytorch/pull/101528/commits/ceeb242bc7a8291c2f108659e46fd81d418380ec

This broke CI pipelines running external dependencies that expected `get_a_var`'s name to not change. In this PR, we change the name back to `get_a_var` and include it in the `__all__` instead.

### CODE REVIEW

**Comment by ringohoffman** [comment]
@ezyang this is ready with the renaming of `_get_a_var` back to `get_a_var`

**Review by ezyang — APPROVED** [review]

**Comment by ringohoffman** [comment]
@ezyang can you add the [topic: not user facing](https://github.com/pytorch/pytorch/labels/topic%3A%20not%20user%20facing) label and try to merge again?

**Comment by ringohoffman** [comment]
@ezyang sorry to double ping, but could we add the tag and try to merge this again?
